src/uct/api/v2/uct_v2.h

```diff
@@ -240,6 +251,18 @@ typedef enum {
 } uct_md_mem_dereg_flags_t;
 
 
+typedef enum {
+    /**
+     * Indicates that remote access to the returned remote key should fail if
+     * @ref uct_md_mem_dereg_v2 is called with the flag
+     * @ref UCT_MD_MEM_DEREG_FLAG_INVALIDATE on the memory handle. If this flag
+     * is not set, using the remote key after invalidating the memory region has
+     * an undefined behavior.
+     */
```

Comment:
yes

Reply:
The flag is used indicate that remote access to a memory region associated with the remote key must fail once the memory region is deregister using @ref uct_md_mem_dereg_v2 and UCT_MD_MEM_DEREG_FLAG_INVALIDATE flag set. Using @ref uct_md_mem_dereg_v2  deregistration routine  with UCT_MD_MEM_DEREG_FLAG_INVALIDATE flag set while the rkey was generated without UCT_MD_MKEY_PACK_FLAG_INVALIDATE flag set may result in data corruption. In other words in order for UCT_MD_MEM_DEREG_FLAG_INVALIDATE flag to function the UCT_MD_MKEY_PACK_FLAG_INVALIDATE flag must be set.

Reply:
@yosefe On a bit different topic. While I was reviewing `uct_md_mem_dereg_v2` I have noticed that `memh` that is non-optional parameter is passed through param/flag. I think for consistency reason it would be useful to move it to arg list.

Reply:
> The flag is used indicate that remote access to a memory region associated with the remote key must fail once the memory region is deregister using @ref uct_md_mem_dereg_v2 and UCT_MD_MEM_DEREG_FLAG_INVALIDATE flag set. Using @ref uct_md_mem_dereg_v2 deregistration routine with UCT_MD_MEM_DEREG_FLAG_INVALIDATE flag set while the rkey was generated without UCT_MD_MKEY_PACK_FLAG_INVALIDATE flag set may result in data corruption. In other words in order for UCT_MD_MEM_DEREG_FLAG_INVALIDATE flag to function the UCT_MD_MKEY_PACK_FLAG_INVALIDATE flag must be set.

applied